### PR TITLE
Add region filters to Megadrive/Genesis when listing available roms

### PR DESCRIPTION
**Pull request by csvowles**
TLDR summary: Do not list non working (region locked) combinations of Megadrive/Genesis console and rom.

The megadriv.xml software list has now been updated to document the region locking of the various ROMs that implement this (https://github.com/mamedev/mame/pull/9688 and https://github.com/mamedev/mame/pull/9732, released in Mame version 0.245).

The code change in this PR would prevent listing a ROM which doesn't work on the regional variant of the Mega Drive / Genesis machine chosen.

Some Mega Drive/Genesis ROMs are region protected, when the game starts they either display a message (explaining the Cartridge is designed for a different region), or they hang (e.g. some Konami games); Either way they are not playable on a machine with an invalid region.

This enhancement filters on the softwarelist to use the optional "compatibility" value set to prevent invalid combinations of Console region and ROM region. It is based on how the 32x Mega Drive based Mame machines achieve this.

If compatibility region is not set for a ROM then it defaults to being listed as available for all 3 consoles (NTSC-U, NTSC-J and PAL) as before.

This PR is a replacement for https://github.com/mamedev/mame/pull/9620 given it has gone stale, and all the changes to directory structure and file naming since that PR was initially raised.

### CODE REVIEW

**Comment by csvowles** [comment]
Rebased due to directory rename `src/mame/segacons` => `src/mame/sega`

**Comment by simzy39** [comment]
Is this ready to merge?

**Comment by csvowles** [comment]
> Is this ready to merge?

As far as I am concerned it is; I compiled and tested it back in June and have been using it locally ever since fine, it filters the ROMs correctly.

**Review by angelosa — APPROVED** [review]

**Review by cuavas — COMMENTED** [review]
I’m not sure this will work properly.  In particular:
* `ms_megadriv2` (not used directly but used as the basis for `ms_megajet`) instantiates the cartridge software list, but you haven’t added a compatibility filter.
* `genesis_tmss` sets the software list filter to `"TMSS"` so the cartridges that lack the TMSS header should be recognised as incompatible.  However, this will replace the region filter from `ms_megadriv` so TMSS-compatible cartridges will be shown as compatible irrespective or regional compatibility.

So I think this won’t work properly for the `genesis_tmss` and `megajet` machines.

**Comment by cuavas** [comment]
See a37910b30edd2abd31bb7546ff50e8859186b9a7.
